test/model/src/core/kv_transaction.cpp

```diff
@@ -111,14 +164,29 @@ kv_transaction::rollback()
 }
 
 /*
- * kv_transaction::set_timestamp --
- *     Set the timestamp for all subsequent updates.
+ * kv_transaction::set_commit_timestamp --
+ *     Set the commit timestamp for all subsequent updates.
  */
 void
-kv_transaction::set_timestamp(timestamp_t commit_timestamp)
+kv_transaction::set_commit_timestamp(timestamp_t commit_timestamp)
 {
     std::lock_guard lock_guard(_lock);
+
+    if (commit_timestamp == k_initial_commit_timestamp)
+        throw model_exception("Invalid commit timestamp");
+
+    kv_transaction_state txn_state = state();
+    if (txn_state != kv_transaction_state::in_progress &&
+      txn_state != kv_transaction_state::prepared)
+        throw model_exception("The transaction must be either in progress or prepared");
```

Comment:
Do you think it'd be worth factoring these out into something like `assert_valid_transition(state old, state new)`? It gets a little repetitive having to think about all possible combinations in several functions.

Reply:
I refactored this check into `assert_in_progress_or_prepared()`, as this is the most common case. I don't think refactoring state transition checks outside the relevant function would make sense so that we can, for example, keep all commit logic together.

Reply:
Fair enough - thanks.